src/internal/operators/take.ts

```diff
@@ -43,51 +43,42 @@ import { MonoTypeOperatorFunction, TeardownLogic } from '../types';
  *
  * @throws {ArgumentOutOfRangeError} When using `take(i)`, it delivers an
  * ArgumentOutOrRangeError to the Observer's `error` callback if `i < 0`.
- *
- * @param {number} count The maximum number of `next` values to emit.
- * @return {Observable<T>} An Observable that emits only the first `count`
+ * @throws {TypeError} when no numeric argument is passed.
+ * @param count The maximum number of `next` values to emit.
+ * @return An Observable that emits only the first `count`
  * values emitted by the source Observable, or all of the values from the source
  * if the source emits fewer than `count` values.
- * @method take
- * @owner Observable
  */
 export function take<T>(count: number): MonoTypeOperatorFunction<T> {
-  return (source: Observable<T>) => {
-    if (count === 0) {
-      return empty();
-    } else {
-      return source.lift(new TakeOperator(count));
-    }
-  };
+  if (isNaN(count)) {
```

Comment:
I think we should use `Number.isNaN` here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

Reply:
Well, not really, `Number.isNaN` would only return true if it were *actually* `NaN`. We want to account for strings that are valid numbers as well. There are people that use RxJS with plain JavaScript (like psychopaths!) and this should be valid for them:

```js
const input = document.querySelector('input[type="text"]');

of(1, 2, 3, 4).pipe(take(input.value)).subscribe(x => console.log(x));
```

DOES THAT CODE SUCK? Yes. Yes it does. But JS has coercion as a feature. 🤷‍♂